cpp/ql/lib/semmle/code/cpp/rangeanalysis/SimpleRangeAnalysis.qll

```diff
@@ -1565,9 +1588,9 @@ private module SimpleRangeAnalysisCached {
    */
   cached
   float upperBound(Expr expr) {
```

Comment:
Do we need to do a similar trick to improve `lowerBound`?

Reply:
Lower bound is slight more complicated due to signed vs unsigned comparison. 
 Consider the  case:
```c
void signed_test(short ss) {
  if (ss >= sizeof(int)) { 
    out(ss); // -32768 .. 32767
  }
```
(Also see comment in the tests here: https://github.com/github/codeql/blob/c20cf23c98a2992908225b57904bc43293435b49/cpp/ql/test/library-tests/rangeanalysis/SimpleRangeAnalysis/test.c#L611).

Implementing lower bound symmetricly to upper bound, did not handle the above case correctly (I tried of course), so I abandoned the idea for now. 

This PR is related to increasing precision on OverflowStatic test, which IIRC only uses upper bound. Of course if there is a simple way to implement lower bound, we should implement it now, but with my limited experience I cannot judge how much effort is needed for this.

Reply:
Ah, right. I see what you mean. The handling of signedness vs. unsignedness is already done in `getTruncatedLowerBounds`, but that work would also need to be done in a `getGuardedLowerBound` predicate if we were to implement this trick for lower bounds as well. I'm totally fine with not doing this in this PR. 👍